src/appsync.ts

```diff
@@ -152,419 +164,447 @@ export class AppsyncVTL extends VTL {
-              templates.push(requestMappingTemplateString);
-              templates.push(responseMappingTemplate);
-              template = new AppsyncVTL(AppsyncVTL.CircuitBreaker);
-              const name = getUniqueName(
-                api,
-                appsyncSafeName(integration.kind)
-              );
-              return dataSource.createFunction({
-                name,
-                requestMappingTemplate: appsync.MappingTemplate.fromString(
-                  requestMappingTemplateString
-                ),
-                responseMappingTemplate: appsync.MappingTemplate.fromString(
-                  responseMappingTemplate
-                ),
-              });
+              // this catches specific cases:
+              // lambdaFunction()
+              // table.get()
+              // table.<method-name>()
+
+              // all other Calls or PropAccessExpr are considered "after" the service call, e.g.
+              // lambdaFunction().prop
+              // table.query().Items.size() //.size() is still a Call but not a call to a service.
+              return expr;
+            } else if (isPropAccessExpr(expr)) {
+              return findServiceCallExpr(expr.expr);
+            } else if (isCallExpr(expr)) {
+              return findServiceCallExpr(expr.expr);
+            } else {
+              throw new Error("");
             }
-          } else if (isLastExpr) {
-            if (isReturnStmt(stmt)) {
-              template.return(stmt.expr);
-            } else if (isIfStmt(stmt)) {
-              template.eval(stmt);
+          }
+
+          /**
+           * Resolve a VTL expression that applies in-line expressions such as PropAccessExpr to
+           * the result from `$context.result` in a Response Mapping template.
+           *
+           * Example:
+           * ```ts
+           * const items = table.query().items
+           * ```
+           *
+           * The Response Mapping Template will include:
+           * ```
+           * #set($context.stash.items = $context.result.items)
+           * ```
+           *
+           * @param expr the {@link Expr} which is triggering a call to an external service, such as a Table or Function.
+           * @returns a VTL expression to be included in the Response Mapping Template to extract the contents from `$context.result`.
+           */
+          function getResult(expr: Expr): string {
+            if (isCallExpr(expr)) {
+              template.call(expr);
+              return returnValName;
+            } else if (isPropAccessExpr(expr)) {
+              return `${getResult(expr.expr)}.${expr.name}`;
+            } else if (isElementAccessExpr(expr)) {
+              return `${getResult(expr.expr)}[${getResult(expr.element)}]`;
             } else {
-              template.return("$null");
+              throw new Error(
+                `invalid Expression in-lined with Service Call: ${expr.kind}`
+              );
```

Comment:
Synth error, also I don't understand how this happens?

Reply:
I haven't changed this code. It is a bit hacky - it is trying to support various different ways in which the result of an Integration call may be referenced.

E.g. 

```ts
call().result
call().result[0]
```

This will be fixed with your a-normal form change? https://github.com/functionless/functionless/pull/280

Reply:
Ohh, yeah, I actually removed this whole function, lol

Reply:
in the async change.